<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WaitForPropertyWidget</class>
 <widget class="QWidget" name="WaitForPropertyWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string comment="@title">Edit property to wait for</string>
  </property>
  <property name="whatsThis">
   <string comment="@info:whatsthis">&lt;para&gt;Set the object name, the property name and the value to wait for.&lt;/para&gt;</string>
  </property>
  <layout class="QVBoxLayout" name="WaitForPropertyVerticalLayout">
   <item>
    <widget class="QGroupBox" name="waitForPropertyGroupBox">
     <property name="title">
      <string comment="@title:group">Wait for property</string>
     </property>
     <layout class="QHBoxLayout" name="waitForPropertyGroupBoxHorizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="labelVerticalLayout">
        <item>
         <widget class="QLabel" name="objectNameLabel">
          <property name="text">
           <string comment="@label:textbox">Object name:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>objectNameLineEdit</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="propertyNameLabel">
          <property name="text">
           <string comment="@label:textbox">Property name:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>propertyNameLineEdit</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="valueLabel">
          <property name="text">
           <string comment="@label:textbox">Value:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>valueLineEdit</cstring>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="valueVerticalLayout">
        <item>
         <widget class="KLineEdit" name="objectNameLineEdit">
          <property name="whatsThis">
           <string comment="@info:whatsthis">&lt;para&gt;The name of the QObject that contains the property.&lt;/para&gt;
&lt;para&gt;Note that the name is not the class of the object, but the string returned by its objectName() method.&lt;/para&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KLineEdit" name="propertyNameLineEdit">
          <property name="whatsThis">
           <string comment="@info:whatsthis">&lt;para&gt;The name of the property.&lt;/para&gt;
&lt;para&gt;Not every property can be used to wait until its value changes to the expected one. Only properties that have a notify signal can be used with that purpose.&lt;/para&gt;
&lt;para&gt;Properties that do not have a notify signal can still be used to enrich other conditions (for example, waiting for a signal to be emitted but only changing to the next step if, in addition, some property has some value), but they can not be used alone as single conditions in a reaction.&lt;/para&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KLineEdit" name="valueLineEdit">
          <property name="whatsThis">
           <string comment="@info:whatsthis">&lt;para&gt;The expected value of the property.&lt;/para&gt;
&lt;para&gt;The value is copied as is to the script. That means that it has to be written with the syntax of the script language that the tutorial is going to be exported to. For example, if the tutorial is going to be exported to JavaScript and the value to wait for is a text, the value has to be written as &lt;em&gt;&quot;The expected text&quot;&lt;/em&gt; (wrapped in quotes).&lt;/para&gt;
&lt;para&gt;The value can even contain a programming expression in the script language. For example, something like &lt;em&gt;t.i18nc(&quot;@item:inlistbox&quot;, &quot;Certain item text&quot;)&lt;/em&gt; would be used to wait for the localized text of an item in a combo box.&lt;/para&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="waitForPropertyWidgetSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
